services/motion/builtin/move_request.go

```diff
@@ -214,6 +302,34 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		}
 	}
 
+	var obstacleDetectors []map[resource.Name]vision.Service
+	for _, obstacleDetectorNamePair := range motionCfg.ObstacleDetectors {
+		pair := make(map[resource.Name]vision.Service)
```

Comment:
Yeah again I think we can make this just a map and not a slice of maps.  Each map entry is a key value pair in itself